tests/test_port_config.py

```diff
@@ -154,6 +155,76 @@ def test_port_breakout(self, dvs, port_config):
             assert hw_lane_value, "Can't get hw_lane list"
             assert hw_lane_value == "1:%s" % (new_lanes[i])
 
+    def test_recirc_port(self, dvs):
+
+        # Get port config from configDB
+        cfg_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        cfg_port_tbl = swsscommon.Table(cfg_db, swsscommon.CFG_PORT_TABLE_NAME)
+
+        indexes = []
+        lanes = []
+        keys = cfg_port_tbl.getKeys()
+        for port in keys:
+            (status, fvs) = cfg_port_tbl.get(port)
+            assert(status == True)
+
+            for fv in fvs:
+                if fv[0] == "index":
+                    indexes.append(int(fv[1]))
+                if fv[0] == "lanes":
+                    lanes.extend([int(lane) for lane in fv[1].split(",")])
+
+        # Stop swss before modifing the configDB
+        dvs.stop_swss()
+        time.sleep(1)
+
+        recirc_port_lane_base = max(lanes) + 1
+        recirc_port_index_base = max(indexes) + 1
+
+        # Add recirc ports to port config in configDB
+        recirc_port_lane_name_map = {}
+        for i in range(2):
+            name = alias = "Ethernet-Rec%s" % i
+            fvs = swsscommon.FieldValuePairs([("role", "Rec" if i % 2 == 0 else "Inb"),
+                                              ("alias", alias),
+                                              ("lanes", str(recirc_port_lane_base + i)),
+                                              ("speed", "10000"),
+                                              ("index", str(recirc_port_index_base + i))])
+            cfg_port_tbl.set(name, fvs)
+
+        # Start swss
+        dvs.start_swss()
+        time.sleep(5)
+
+        polling_config = PollingConfig(polling_interval=0.1, timeout=15, strict=True)
+
+        # Verify recirc ports in port table in applDB
+        for i in range(2):
+            name = alias = "Ethernet-Rec%s" % i
```

Comment:
same as above

Reply:
@abdosi The name do not really matter here. What matters is the port role. We covered both port roles: Rec and Inb in the test (Line188 and 205).